python/mxnet/gluon/block.py

```diff
@@ -199,6 +198,32 @@ def __setattr__(self, name, value):
 
         super(Block, self).__setattr__(name, value)
 
+    def _check_container_with_block(self):
+        def _find_block_in_container(data):
+            # Find whether a nested container structure contains Blocks
+            if isinstance(data, (list, tuple)):
+                for ele in data:
+                    if _find_block_in_container(ele):
+                        return True
+                return False
+            elif isinstance(data, dict):
+                for _, v in data.items():
+                    if _find_block_in_container(v):
+                        return True
+                return False
+            elif isinstance(data, Block):
+                return True
+            else:
+                return False
+        for k, v in self.__dict__.items():
+            if isinstance(v, (list, tuple, dict)) and not (k.startswith('__') or k == '_children'):
+                if _find_block_in_container(v):
+                    warnings.warn('Find "{name}" to be a container with Blocks. '
+                                  'Note that Blocks inside the list, tuple or dict will not be '
+                                  'registered automatically! Make sure to register them using '
+                                  'register_child() or use nn.Sequential instead.'
```

Comment:
what if it's a hybrid block?

Reply:
I've revised the message to be "nn.Sequential/nn.HybridSequential`. Also, the HybridBlock is inherited from Block and it should be okay to just add the check here.